addons/mail/static/src/core/common/message_actions.js

```diff
@@ -201,7 +201,8 @@ messageActionsRegistry
     .add("copy-link", {
         condition: (component) =>
             component.message.message_type &&
-            component.message.message_type !== "user_notification",
+            component.message.message_type !== "user_notification" &&
+            (!component.props.thread.access_token || component.props.thread.hasReadAccess),
```

Comment:
Please make a separate PR for this small part as we can merge this independently on the discussion about the python part.